agents-common/src/main/java/org/apache/ranger/authorization/hadoop/config/RangerLegacyConfigBuilder.java

```diff
@@ -19,227 +19,220 @@
 
 package org.apache.ranger.authorization.hadoop.config;
 
-import java.io.File;
-import java.net.URL;
-import java.util.HashMap;
-import java.util.Map;
-
 import org.apache.hadoop.conf.Configuration;
 import org.apache.ranger.plugin.store.EmbeddedServiceDefsUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.File;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
 public class RangerLegacyConfigBuilder {
+    private static final Logger LOG = LoggerFactory.getLogger(RangerLegacyConfigBuilder.class);
+
+    static String serviceType;
+    static String legacyResource;
+
+    private RangerLegacyConfigBuilder() {
+        // to block instantiation
+    }
+
+    public static Configuration getSecurityConfig(String serviceType) {
+        RangerLegacyConfigBuilder.legacyResource = getPropertyName(RangerConfigConstants.XASECURE_SECURITY_FILE, serviceType);
+        RangerLegacyConfigBuilder.serviceType    = serviceType;
+
+        Configuration ret           = null;
```

Comment:
nit: null can be removed.

Reply:
Removing initialization in line 46 will result in compilation error, since ret will not assigned before referencing at line 60.